sbin/autograder/packer_unpacker.py

```diff
@@ -0,0 +1,301 @@
+    obj["queue_time"] = queue_time_longstring
+    obj["regrade"] = is_batch_job
+    obj["waittime"] = waittime
+    obj["job_id"] = job_id
+
+    with open(os.path.join(tmp_submission,"queue_file.json"),'w') as outfile:
+        json.dump(obj,outfile,sort_keys=True,indent=4,separators=(',', ': '))
+
+    grading_began_longstring = dateutils.write_submitty_date(grading_began)
+    with open(os.path.join(tmp_submission,".grading_began"), 'w') as f:
+        print (grading_began_longstring,file=f)
+
+    # zip up autograding & submission folders
+    filehandle1, my_autograding_zip_file =tempfile.mkstemp()
+    filehandle2, my_submission_zip_file =tempfile.mkstemp()
+    grade_item.zip_my_directory(tmp_autograding,my_autograding_zip_file)
+    grade_item.zip_my_directory(tmp_submission,my_submission_zip_file)
+    os.close(filehandle1)
+    os.close(filehandle2)
+    # cleanup
+    shutil.rmtree(tmp_autograding)
+    shutil.rmtree(tmp_submission)
+    shutil.rmtree(tmp)
+
+    #grade_items_logging.log_message(job_id,is_batch_job,"done zip",item_name)
+
+    return (my_autograding_zip_file,my_submission_zip_file)
+
+
+# ==================================================================================
+# ==================================================================================
+def unpack_grading_results_zip(which_machine,which_untrusted,my_results_zip_file):
+    os.chdir(SUBMITTY_DATA_DIR)
+
+    queue_obj = unzip_queue_file(my_results_zip_file)
+
+    job_id = queue_obj["job_id"]
+    partial_path = os.path.join(queue_obj["gradeable"],queue_obj["who"],str(queue_obj["version"]))
+    item_name = os.path.join(queue_obj["semester"],queue_obj["course"],"submissions",partial_path)
+    results_path = os.path.join(SUBMITTY_DATA_DIR,"courses",queue_obj["semester"],queue_obj["course"],"results",partial_path)
+
+    # clean out all of the old files if this is a re-run
+    shutil.rmtree(results_path,ignore_errors=True)
+    # create the directory (and the full path if it doesn't already exist)
+    os.makedirs(results_path)
+    # unzip the file & clean up
+    grade_item.unzip_this_file(my_results_zip_file,results_path)
+    os.remove(my_results_zip_file)
+
+    # add information to the database
+    insert_database_version_data.insert_to_database(
+        queue_obj["semester"],
+        queue_obj["course"],
+        queue_obj["gradeable"],
+        queue_obj["user"],
+        queue_obj["team"],
+        queue_obj["who"],
+        True if queue_obj["is_team"] else False,
+        str(queue_obj["version"]))
+
+    submission_path = os.path.join(SUBMITTY_DATA_DIR,"courses",item_name)
+
+    is_batch_job = queue_obj["regrade"]
+    gradingtime=queue_obj["gradingtime"]
+    grade_result=queue_obj["grade_result"]
+
+    print (which_machine,which_untrusted,"unzip",item_name, " in ", int(gradingtime), " seconds")
+
+    grade_items_logging.log_message(job_id,is_batch_job,"unzip",item_name,"grade:",gradingtime,grade_result)
```

Comment:
I dislike the fact that a module named `packer_unpacker.py` is when asked to unpack a gradeable is via `unpack_grading_results_zip`:
1. unpacking the gradeable details
1. writing the gradeable data to the database
1. printing a message to the console about grading time
1. logging the details of the grading

which isn't great. This isn't really a thing we need to change for this PR, but we really should try and improve the naming and structure of the autograder code at some point.

Reply:
Can you be more specific about what isn't great?  Is it doing too much?  I think it makes sense.  
(But yes, let's hold off on whatever change you would like to see for another issue/PR)

Reply:
I'd say it's doing too much for what the function and module are named. I'd assume a module named packer_unpacker.py would only deal with packing and unpacking zips.